ext/sodium/libsodium.c

```diff
@@ -0,0 +1,2682 @@
+						   crypto_pwhash_scryptsalsa208sha256_memlimit_interactive(), CONST_CS | CONST_PERSISTENT);
+	REGISTER_LONG_CONSTANT("SODIUM_CRYPTO_PWHASH_SCRYPTSALSA208SHA256_OPSLIMIT_SENSITIVE",
+						   crypto_pwhash_scryptsalsa208sha256_opslimit_sensitive(), CONST_CS | CONST_PERSISTENT);
+	REGISTER_LONG_CONSTANT("SODIUM_CRYPTO_PWHASH_SCRYPTSALSA208SHA256_MEMLIMIT_SENSITIVE",
+						   crypto_pwhash_scryptsalsa208sha256_memlimit_sensitive(), CONST_CS | CONST_PERSISTENT);
+	REGISTER_LONG_CONSTANT("SODIUM_CRYPTO_SCALARMULT_BYTES",
+						   crypto_scalarmult_BYTES, CONST_CS | CONST_PERSISTENT);
+	REGISTER_LONG_CONSTANT("SODIUM_CRYPTO_SCALARMULT_SCALARBYTES",
+						   crypto_scalarmult_SCALARBYTES, CONST_CS | CONST_PERSISTENT);
+	REGISTER_LONG_CONSTANT("SODIUM_CRYPTO_SHORTHASH_BYTES",
+						   crypto_shorthash_BYTES, CONST_CS | CONST_PERSISTENT);
+	REGISTER_LONG_CONSTANT("SODIUM_CRYPTO_SHORTHASH_KEYBYTES",
+						   crypto_shorthash_KEYBYTES, CONST_CS | CONST_PERSISTENT);
+	REGISTER_LONG_CONSTANT("SODIUM_CRYPTO_SECRETBOX_KEYBYTES",
+						   crypto_secretbox_KEYBYTES, CONST_CS | CONST_PERSISTENT);
+	REGISTER_LONG_CONSTANT("SODIUM_CRYPTO_SECRETBOX_MACBYTES",
+						   crypto_secretbox_MACBYTES, CONST_CS | CONST_PERSISTENT);
+	REGISTER_LONG_CONSTANT("SODIUM_CRYPTO_SECRETBOX_NONCEBYTES",
+						   crypto_secretbox_NONCEBYTES, CONST_CS | CONST_PERSISTENT);
+	REGISTER_LONG_CONSTANT("SODIUM_CRYPTO_SIGN_BYTES",
+						   crypto_sign_BYTES, CONST_CS | CONST_PERSISTENT);
+	REGISTER_LONG_CONSTANT("SODIUM_CRYPTO_SIGN_SEEDBYTES",
+						   crypto_sign_SEEDBYTES, CONST_CS | CONST_PERSISTENT);
+	REGISTER_LONG_CONSTANT("SODIUM_CRYPTO_SIGN_PUBLICKEYBYTES",
+						   crypto_sign_PUBLICKEYBYTES, CONST_CS | CONST_PERSISTENT);
+	REGISTER_LONG_CONSTANT("SODIUM_CRYPTO_SIGN_SECRETKEYBYTES",
+						   crypto_sign_SECRETKEYBYTES, CONST_CS | CONST_PERSISTENT);
+	REGISTER_LONG_CONSTANT("SODIUM_CRYPTO_SIGN_KEYPAIRBYTES",
+						   crypto_sign_SECRETKEYBYTES + crypto_sign_PUBLICKEYBYTES,
+						   CONST_CS | CONST_PERSISTENT);
+	REGISTER_LONG_CONSTANT("SODIUM_CRYPTO_STREAM_NONCEBYTES",
+						   crypto_stream_NONCEBYTES, CONST_CS | CONST_PERSISTENT);
+	REGISTER_LONG_CONSTANT("SODIUM_CRYPTO_STREAM_KEYBYTES",
+						   crypto_stream_KEYBYTES, CONST_CS | CONST_PERSISTENT);
+	return SUCCESS;
+}
+
+PHP_MSHUTDOWN_FUNCTION(libsodium)
+{
+	return SUCCESS;
+}
+
+PHP_MINFO_FUNCTION(libsodium)
+{
+	php_info_print_table_start();
+	php_info_print_table_header(2, "libsodium support", "enabled");
+	php_info_print_table_header(2, "libsodium compiled version", PHP_LIBSODIUM_VERSION);
+	php_info_print_table_header(2, "libsodium headers version", SODIUM_VERSION_STRING);
+	php_info_print_table_header(2, "libsodium library version", sodium_version_string());
+	php_info_print_table_end();
+}
+
+PHP_FUNCTION(sodium_memzero)
+{
+	zval	  *buf_zv;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(),
+							  "z", &buf_zv) == FAILURE) {
```

Comment:
wouldn't be better to indent it with less tabs? I think that 2 tabs (3 with the function indent) should be enough.